### PR TITLE
web3-eth exporting of types from web3-core

### LINKED ISSUE #3197: web3-eth should export TransactionConfig
Hello!

I'm using `sendTransaction` method from the `web3-eth` package. The first argument has a type `TransactionConfig`, however, this type is not exported from `web3-eth`. I have to additionally and explicitly install the `web3-core` package just to import `TransactionConfig` from it. This is very cumbersome and creates tight coupling between modules.

You should re-export `TransactionConfig` from `web3-eth` package or even to define it in the package itself to decouple packages more (if it makes sense).

Thank you!

### PR DESCRIPTION
**Pull request by nivida**
## Description

This PR will improve the DX for TS developers. 

Fixes #3197 

## Type of change

- [x] Types

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run test``` with success and extended the tests if necessary.
- [x] I ran ```npm run build``` and tested the resulting file from ```dist``` folder in a browser.
- [ ] I have tested my code on the live network.

### CODE REVIEW

**Review by cgewecke — COMMENTED** [review]
I'm seeing the truffle e2e CI fail on `tsc` because it can't find `Block` exported from web3-eth. 
```
$ tsc
lib/interface-adapter/types/index.d.ts:1:10 - error TS2305: Module '"../../../node_modules/web3-eth/types"' has no exported member 'Block'.
1 import { Block as EvmBlock } from "web3-eth";
           ~~~~~
Found 1 error.
```

It looks like it was removed in [this commit](https://github.com/ethereum/web3.js/commit/73f338279655a108af9952926ca28009083a66f5#diff-db8d4b5a4b9c1b6661a28c635c5b062bL361). 

Is there any way of restoring that as an alias in this PR? 

See also #3150

**Comment by nivida** [comment]
> Is there any way of restoring that as an alias in this PR?

Added the ``Block`` interface with a ``TODO`` comment to remove this type on a minor release.

File: packages/web3-eth/types/tests/eth.tests.ts [hunk 1]
```diff
@@ -18,13 +18,22 @@
  * @date 2018
  */
 
+import {Log} from 'web3-core';
 import {
```
Comment (joshstevens19) [inline packages/web3-eth/types/tests/eth.tests.ts:22]: why are you importing them when its not being tested?

**Review by joshstevens19 — APPROVED** [review]

File: packages/web3-eth/types/tests/eth.tests.ts (hunk 1 above)
Comment (nivida) [reply packages/web3-eth/types/tests/eth.tests.ts:22]: I have imported them as a test. This should give us the proof those forwarded types do exist in the ``web3-eth`` package. :)

**Pull request merged**